_source/_posts/2022-10-01-quick-javascript-login.adoc

```diff
@@ -0,0 +1,166 @@
+---
+layout: blog_post
+title: "Quick JavaScript Login with Schematics"
+author: matt-raible
+by: advocate
+communities: [javascript]
+description: ""
+tags: []
+tweets:
+- ""
+- ""
+- ""
+image: blog/oktadev-schematics-auth0/quick-javascript-login.jpg
+type: conversion
+---
+:page-liquid:
+:toc: macro
+:experimental:
+
+// Other title ideas:
+//   - Quick JavaScript Login with Schematics (current, 58)
+//   - Schematics for Quick JavaScript Authentication (59)
+//   - Quick Templates for JavaScript Authentication (59)
+//   - CLI to App to Auth in Minutes with OktaDev Schematics! (60)
+
+// Keywords:
+//   - javascript authentication: 480 (there's an Okta post on the first page)
+//   - javascript auth: 210
+//   - javascript login: 880
+//   - javascript security: 720
+//   - javascript frameworks: 27,100
+//   - react oidc: 720
+//   - oidc authentication: 2900
+//   - jwt auth: 2400
+//   - jwt javascript: 720
+
+// Should we emphasize this is a developer story moreso than a product pitch?
+
+Hello,
+
+My name is Matt Raible, and I am a developer advocate at Okta. I've worked here for over five years; before that, I was an independent consultant for 20 years, mainly doing Java and web development for clients.
+
+I've learned a lot about OpenID Connect (OIDC) and Okta's SDKs over my Okta years and only recently tried to use Auth0's. Over the last couple of weeks, I integrated many Auth0 SDKs into OktaDev Schematics.
+
+I was pleasantly surprised by the experience because the configuration often boils down to two values: `domain` and `clientId`. And then, it just works! There's no need to worry about issuers, redirect URIs, or callback URLs.
+
+This is a strong departure from how we've done things in Okta's SDKs, where we expect people to know about OpenID Connect. And from a developer advocacy standpoint, we often try to educate developers about OAuth and OIDC. Since these topics are complex, they can be challenging to understand. I like the approach of just needing two values, and everything just works!
```

Comment:
I'll ask someone else to review this paragraph. It has some negative connotation about Okta, to which I'm not 100% sure about, perhaps it can be rewritten.

Reply:
I agree with @bajcmartinez, it suggests that Okta SDKs don’t do their job well. Even if it's the case, I don't think it's a good thing to admit it publicly 😂 
Maybe it could be reformulated as follows:

"We often try to educate developers about OAuth and OIDC but since these topics are complex, they can be challenging to understand. I like the approach of just needing two values, and everything just works!"

What do you think?

Reply:
I like it! I've added this as a suggestion.